.travis.yml

```diff
@@ -81,6 +85,17 @@ jobs:
         # Note $behind is NOT comparable to clone depth in repos with merges.
         - '[ "$behind" -lt 1000 ] # is the PR base too far behind?'
 
+    - name: checkpatch
+      script:
+        # If git misses this merge base then 'git rev-list
+        # ^${TRAVIS_BRANCH} ${TRAVIS_PULL_REQUEST_SHA}' will tell
+        # checkpatch to scan practically every single git commit in the
+        # repo.
+        #
+        # FIXME: this addresses only pull requests without any merge
+        # submitted as part of the pull request itself.
+        - *unshallow_for_merge_base
```

Comment:
out of curiosity, what are those '*unshallow_for_merge_base" and "&unshallow_for_merge_base"?

Reply:
They're YAML anchors = how to avoid copy/paste/diverge.